lib/speech/index.js

```diff
@@ -0,0 +1,903 @@
+ * @param {number} streamingRecognizeRequest.streamingConfig.config.sampleRate
+ *     - Sample rate in Hertz of the audio data sent in all RecognitionAudio
+ *     messages. See
+ *     [`RecognitionConfig.sampleRate`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ * @param {string=}
+ *     streamingRecognizeRequest.streamingConfig.config.languageCode
+ *     - See [`RecognitionConfig.languageCode`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `"en-US"`.
+ * @param {number=}
+ *     streamingRecognizeRequest.streamingConfig.config.maxAlternatives
+ *     - See [`RecognitionConfig.maxAlternatives`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `1`.
+ * @param {boolean=}
+ *     streamingRecognizeRequest.streamingConfig.config.profanityFilter
+ *     - See [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=}
+ *     streamingRecognizeRequest.streamingConfig.config.speechContext
+ *     - See [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {boolean=} streamingRecognizeRequest.streamingConfig.singleUtterance
+ *     - See [`StreamingRecognitionConfig.singleUtterance`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {boolean=} streamingRecognizeRequest.streamingConfig.interimResults
+ *     - See [`StreamingRecognitionConfig.interimResults`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.StreamingRecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @returns {object} Readable and writable stream.
+ *
+ * @example
+ * var fs = require('fs');
+ * var Transform = require('stream').Transform;
+ * var toRecognizeRequest = new Transform();
+ * toRecognizeRequest._transform = function (chunk, encoding, done) {
+ *   done(null, {
+ *     audioContent: chunk
+ *   });
+ * };
+ *
+ * // See the link in the parameters for `StreamingRecognizeRequest`.
+ * var streamingRecognizeRequest = {
+ *   streamingConfig: {
+ *     config: {
+ *       encoding: 'LINEAR16',
+ *       sampleRate: 16000
+ *     },
+ *     singleUtterance: false,
+ *     interimResults: false
+ *   }
+ * };
+ *
+ * var recognizeStream = speech.streamingRecognize(streamingRecognizeRequest)
+ *   .on('data', function (apiResponse) {
+ *     console.log(apiResponse);
```

Comment:
Is `data` the correct event for getting the `apiResponse`? I know we map the `status` event to a `response` event, which I thought was the only way to get an `apiResponse` from a grpc stream.